### PR TITLE
[GLUTEN-1877][CH] Fix bugs when reading hive text

### PR DESCRIPTION
**Pull request by taiyang-li**
## What changes were proposed in this pull request?

(Please fill in changes proposed in this fix)

(Fixes: \#1877)

代码改动: 
- 复用HiveText input format, 去除重复代码
- 修复text格式input_field_names初始化错误导致的查询失败
- 修复count(1)/count(*)查询失败 

## How was this patch tested?

(Please explain how this patch was tested. E.g. unit tests, integration tests, manual tests)

(If this patch involves UI changes, please attach a screenshot; otherwise, remove this)

### CODE REVIEW

**Comment by PHILO-HE** [comment]
Just changed the title to the one more friendly to developers. Please adjust it if you think necessary.

File: cpp-ch/local-engine/Storages/SubstraitSource/TextFormatFile.cpp [hunk 1]
```diff
@@ -1,107 +1,36 @@
...
 
-TextRowInputFormat::TextRowInputFormat(
-    const DB::Block & header_, std::shared_ptr<DB::PeekableReadBuffer> buf_, const DB::RowInputFormatParams & params_, const DB::FormatSettings & format_settings_)
-    : DB::CSVRowInputFormat(
-        header_, buf_, params_, true, false, format_settings_, std::make_unique<TextFormatReader>(*buf_, format_settings_))
-{
-}
-
-void TextRowInputFormat::readPrefix()
-{
```
Comment (KevinyhZou) [inline cpp-ch/local-engine/Storages/SubstraitSource/TextFormatFile.cpp:72]: This readPrefix() can not be removed, it was introduced to slove the problem the field index error,  And you can test the sql `select count(*) from $table where id < ''` can reproduce this

Comment (taiyang-li) [reply cpp-ch/local-engine/Storages/SubstraitSource/TextFormatFile.cpp:72]: column mapping is implemented through hive_text.input_field_names.

**Comment by loneylee** [comment]
LGTM

**Pull request merged**